### PR TITLE
Add Dorgql, dependency, and helper function for test

### PR DESCRIPTION
**Pull request by btracey**

### CODE REVIEW

**Comment by btracey** [comment]
I'll fix the merge confilcts with #91 when that gets an lgtm.

**Comment by btracey** [comment]
Pulled from master. PTAL

File: native/dorgql.go [hunk 1]
```diff
@@ -0,0 +1,100 @@
...
+			iws = n * nb
+			if lwork < iws {
+				// Not enough workspace to use optimal nb: reduce nb and determine
+				// the minimum value of nb.
+				nb = lwork / n
+				nbmin = max(2, impl.Ilaenv(2, "DORGQL", " ", m, n, k, -1))
+			}
+		}
+	}
+	var kk int
```
Comment (kortschak) [inline native/dorgql.go]: Blank line above this.

File: native/dorgql.go [hunk 2]
```diff
@@ -0,0 +1,100 @@
...
+	if lwork == -1 {
+		return
+	}
+	if lwork < n {
+		panic(badWork)
+	}
+	if n == 0 {
+		return
+	}
+	nbmin := 2
```
Comment (kortschak) [inline native/dorgql.go]: Blank line above.

File: native/dorgql.go [hunk 3]
```diff
@@ -0,0 +1,100 @@
...
+	// Use unblocked code for the first or only block.
+	impl.Dorg2l(m-kk, n-kk, k-kk, a, lda, tau, work)
+	if kk > 0 {
+		// Use blocked code.
+		for i := k - kk; i < k; i += nb {
+			ib := min(nb, k-i)
+			if n-k+i > 0 {
+				// Form the triangular factor of the block reflector
+				// H = H[i+ib-1] * ... * H[i+1] * H[i].
+				impl.Dlarft(lapack.Backward, lapack.ColumnWise, m-k+i+ib, ib, a[n-k+i:], lda, tau[i:], work, ldwork)
```
Comment (kortschak) [inline native/dorgql.go]: Split these.

File: native/dorgql.go [hunk 4]
```diff
@@ -0,0 +1,100 @@
...
+
+				// Apply H to A[0:m-k+i+ib, 0:n-k+i] from the left.
+				impl.Dlarfb(blas.Left, blas.NoTrans, lapack.Backward, lapack.ColumnWise, m-k+i+ib, n-k+i, ib, a[n-k+i:], lda, work, ldwork, a, lda, work[ib*ldwork:], ldwork)
+			}
+
+			// Apply H to rows 0:m-k+i+ib of current block.
+			impl.Dorg2l(m-k+i+ib, ib, ib, a[n-k+i:], lda, tau[i:], work)
+
+			// Set rows m-k+i+ib:m of current block to zero.
+			for j := n - k + i; j < n-k+i+ib; j++ {
```
Comment (kortschak) [inline native/dorgql.go]: Ugh. And the fortran is worse for readability.

File: native/dorgql.go (hunk 2 above)
Comment (btracey) [inline native/dorgql.go]: Done.

File: native/dorgql.go (hunk 1 above)
Comment (btracey) [inline native/dorgql.go]: Done.

File: native/dorgql.go (hunk 3 above)
Comment (btracey) [inline native/dorgql.go]: Done.

File: native/dorgql.go (hunk 4 above)
Comment (btracey) [inline native/dorgql.go]: Yea...

**Comment by kortschak** [comment]
LGTM but wait for @vladimir-ch.

File: native/dorg2l.go [hunk 5]
```diff
@@ -0,0 +1,63 @@
...
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/blas/blas64"
+)
+
+// Dorg2l generates an m×n matrix Q with orthenormal columns which is defined
```
Comment (vladimir-ch) [inline native/dorg2l.go]: orthonormal

File: native/dorgql.go [hunk 6]
```diff
@@ -0,0 +1,105 @@
...
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack"
+)
+
+// Dorgql generates the m×n matrix Q with orthonormal columns defined as the
+// last n columns of a product of k elementary reflectors of order m as returned
+// by Dgelqf. That is,
+//  Q = H[k-1] * ... * H[1] * H[0].
+// Please see Dgelqf for more information.
```
Comment (vladimir-ch) [inline native/dorgql.go]: Without 'Please'?

File: native/dorgql.go [hunk 7]
```diff
@@ -0,0 +1,105 @@
...
+// license that can be found in the LICENSE file.
+
+package native
+
+import (
+	"github.com/gonum/blas"
+	"github.com/gonum/lapack"
+)
+
+// Dorgql generates the m×n matrix Q with orthonormal columns defined as the
```
Comment (vladimir-ch) [inline native/dorgql.go]: ... generates _an_ m×n ...?

File: native/dorgql.go [hunk 8]
```diff
@@ -0,0 +1,105 @@
...
+
+	nbmin := 2
+	var nx, ldwork int
+	iws := n
+	if nb > 1 && nb < k {
+		// Determine when to cross over from blocked to unblocked code.
+		nx = max(0, impl.Ilaenv(3, "DORGQL", " ", m, n, k, -1))
+		if nx < k {
+			// Determine if workspace is large enough for blocked code.
+			ldwork = nb
```
Comment (vladimir-ch) [inline native/dorgql.go]: Fortran assigns `n` here, not `nb`.

**Comment by vladimir-ch** [comment]
I still haven't internalized effortless switching between 1-based and 0-based indexing, so with this disclaimer (applied also to any past and future reviews), LGTM

File: native/dorgql.go (hunk 8 above)
Comment (vladimir-ch) [inline native/dorgql.go]: This also appears in the other PR, which cannot be a coincidence. Ignore this comment.

Comment (btracey) [inline native/dorgql.go]: Work is input as a linear array, but it's used as a matrix. Fortran is column-based so the stride is n, while we're row based so the stride is nb.

File: native/dorg2l.go (hunk 5 above)
Comment (btracey) [inline native/dorg2l.go]: Done.

File: native/dorgql.go (hunk 6 above)
Comment (btracey) [inline native/dorgql.go]: Politeness offer revoked

File: native/dorgql.go (hunk 7 above)
Comment (btracey) [inline native/dorgql.go]: I think the is better here. It's a specific matrix that has already been constructed just implicitly defined.

**Pull request merged**